### PR TITLE
Update provision_salt.sh

### PR DESCRIPTION
**Pull request by yusuf-goog**
Updating curl command args in provision_salt.sh to add -L, which allows curl to follow the page if the server says it has moved.

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by yusuf-goog** [comment]
Fixes 92862

File: dev/provision_salt.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ function install_salt() {
     sudo apt update
     sudo apt install salt-minion
   elif [[ "$(uname)" == 'Darwin' ]]; then
-    curl https://repo.saltstack.com/osx/salt-3002.1-py3-x86_64.pkg -o /tmp/salt.pkg
```
Comment (CaseyHillers) [inline dev/provision_salt.sh:22]: From a security standpoint, is it preferred that we update the URL instead of following redirects?

**Comment by CaseyHillers** [comment]
> Fixes 92862

GitHub searches for this information in the PR body, and expects the full https link

File: dev/provision_salt.sh (hunk 1 above)
Comment (yusuf-goog) [reply dev/provision_salt.sh:22]: removed the redirect and renamed the URLs to point to their new saltproject domain name.

@godofredoc for future use, whats the security recommendation on redirects?

**Review by CaseyHillers — APPROVED** [review]
flutter-lgtm

**Pull request merged**